packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -97,6 +104,40 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             new UriAwareCommandHandler(this.selectionService, {
                 execute: (uri: URI) => this.saveAs(uri),
             }));
+        commands.registerCommand(WorkspaceCommands.UPLOAD, new FileSelection.CommandHandler(this.selectionService, {
+            multi: false,
+            isEnabled: selection => this.isEnabled(selection),
+            isVisible: selection => this.isVisible(selection),
+            execute: selection => this.upload(selection)
+        }));
+    }
+
+    protected isEnabled(selection: FileSelection | undefined): boolean {
+        return !environment.electron.is();
+    }
+
+    protected isVisible(selection: FileSelection | undefined): boolean {
+        return !environment.electron.is() && (selection === undefined || selection.fileStat.isDirectory);
+    }
+
+    protected async upload(selection: FileSelection): Promise<void> {
```

Comment:
Can we document this method to describe the default behavior, behavior if a selection is present and so on?

Reply:
of course

Reply:
please use proper types `selection?: FileSelection`, otherwise it is not clear why do you check for `undefined` if it is not expected

Reply:
Can we make UPLOAD command to return FileUploadResult? 
`protected async upload(selection: FileSelection): Promise<FileUploadResult | undefined> {`